Storage/Storage/CoreData/CoreDataIterativeMigrator.swift

```diff
@@ -89,6 +89,10 @@ final class CoreDataIterativeMigrator {
 
         var debugMessages = [String]()
 
+        guard modelsToMigrate.count > 1 else {
+            return (false, [])
+        }
```

Comment:
I can't quite find a scenario that this would fail. I tried running a migration from the same model version and it looks like this is enough to stop the iterative migration: 

https://github.com/woocommerce/woocommerce-ios/blob/4bfcb596ef5507b03e79e6c08a0e3c9b9786c9ec/Storage/Storage/CoreData/CoreDataIterativeMigrator.swift#L45-L49

In any case, it's still a nice addition for protection from ourselves. 😄 

I'd suggest adding an error message to help inform us in Sentry. 

```suggestion
        guard modelsToMigrate.count > 1 else {
            return (false, ["Skipping migration. Unexpectedly found less than 2 models to perform a migration."])
        }
```